File: 2490. Circular Sentence/approach_2.cpp
class Solution {
public:
    bool isCircularSentence(string sentence) {
        bool ans = sentence.back() == sentence[0];
        for (int i = 0; i < (int)sentence.size(); ++i)
            if (sentence[i] == ' ')
                ans &= sentence[i - 1] == sentence[i + 1];
        return ans;
    }
};

/*
    Time: O(n)
    Space: O(1)
*/

File: 1. Two Sum/approach_2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> idx = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            int diff = target - nums[i];
            if (idx.containsKey(diff)) {
                return new int[] {idx.get(diff), i};
            }
            idx.put(nums[i], i);
        }

        return new int[] {-1, -1};
    }
}

/*
    Time Complexity: O(n)
    Space Complexity: O(n)
*/
